### ISSUE TITLE
Bug in tlsio_openssl_create method

### ISSUE THREAD

**Issue by vaskravchuk** [Azure/azure-c-shared-utility#138]
https://github.com/Azure/azure-c-shared-utility/blob/3d1cd1c4740049de47375c5a11aa4c0d6892a040/src/tlsio_openssl.c#L1125

SOCKETIO_CONFIG is created in block code where it will be destroyed, and it's passed outside of block via io_interface_parameters.
```
else
{
   SOCKETIO_CONFIG socketio_config;
   socketio_config.hostname = tls_io_config->hostname;
   socketio_config.port = tls_io_config->port;
  socketio_config.accepted_socket = NULL;

  underlying_io_interface = socketio_get_interface_description();
  io_interface_parameters = &socketio_config;
}
```

**Comment by jspaith** [comment]
Thanks for calling this out & apologies for delay.  

You're absolutely correct; the socketio_config is going out of scope and the fact that the code just works is an accident of nothing assigning over that.

There's similar bugs in our wolfssl and schannel implementations.  I've reviewed rest of SDK but haven't found this other places.

I have a fix prep'd and getting it through our review and gates.  Thanks again @vaskravchuk.

**Comment by jspaith** [comment]
Fixed with commit 86e5bcb9e5438bcdc373800cf6499fc812f8f2b7.
